NeoML/src/Dnn/AutoDiffFunctions.cpp

```diff
@@ -66,11 +68,13 @@ static void getConsequentAxesDimensions( const CDnnBlob* first, const CArray<int
 	}
 }
 
-static bool isConsequentAxes( const CArray<int>& axes )
+static bool isSequentialAxes( const CDnnBlob* blob, const CArray<int>& axes )
 {
 	for( int i = 1; i < axes.Size(); i++ ) {
-		if( axes[i] != axes[i - 1] + 1 ) {

```

Comment:
Infinite loop if `axes[i - 1] > axes[i]`

Reply:
You should somehow guarantee that `axes` contains unique elements in increasing order

Reply:
And in both layers (in Python (`ValueError`) and in C++ (`NeoAssert`))